### ISSUE TITLE
Multi Lingual website

### ISSUE THREAD

**Issue by xiki808** [jbaubree/vite-plugin-pages-sitemap#465]
Can this plugin generate sitemap for a multiple lingual website? If so, how?

**Comment by jbaubree** [comment]
Hi @xiki808 this is not possible for now. I will work on it as soon as possible. If you need this feature quickly, feel free too PR with an i18n config. You can find some inspiration here => [sitemap-ts](https://github.com/jbaubree/sitemap-ts/blob/7dc1990439853100a6c32a9d05e23aa7d0df2b73/src/sitemap.ts#L47)

**Comment by jbaubree** [comment]
@xiki808 you can now use `v1.7.1` using [i18n](https://github.com/jbaubree/vite-plugin-pages-sitemap?tab=readme-ov-file#i18n) config!

**Comment by xiki808** [comment]
Actually I'm not sure how I managed to build the first few times as I'm always getting the following error now:

`TypeError: [vite-plugin-pages] Could not load virtual:vite-plugin-pages/generated-pages?id=~pages (imported by node_modules/@horizon/frontend/src/lib/router.js): Invalid URL`

I guess it's some conflict as I'm actually importing `import routes from '~pages'` inside a custom NPM package that I have which provides basic components and initiates things such as the router itself

**Comment by jbaubree** [comment]
Well, i think you can use `dynamicRoutes` to load your sitemap.
```
onRoutesGenerated: async (routes) => {
    const languages = ['en', 'it']
    const dynamicRoutes = languages.map(lang => `/${lang}/test`)
    generateSitemap({ routes: [...routes, ...dynamicRoutes] })
},
```
I think your TypeError is not depending on this plugin but directly `vite-plugin-pages`. You can maybe add `"skipLibCheck": true` in your tsconfig.json or add `--skipLibCheck` in your tsc command.

**Comment by xiki808** [comment]
Thanks will try it out with the dynamicRoutes.

As for the TypeError, indeed that's what the error is pointing to, but if I just comment out the i18n config line  the projects builds and I also get all the routes.

**Comment by jbaubree** [comment]
Yes sure, feel free to PR a solution for your situation. I will add this to my roadmap but will not get any time for now to develop this feature asap.

**Comment by xiki808** [comment]
As for the TypeError, that's was a silly mistake from my side.. was not returning the routes, `vite-plugin-pages`'s `onRoutesGenerated` function

**Comment by xiki808** [comment]
As for the TypeError, the issue was not returning an asynced function to `vite-plugin-pages`'s `onRoutesGenerated` function
